03_Modules/Ocpi/src/modules/temp/credentials.api.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  AbstractModuleApi,
+  AsDataEndpoint, AuthorizationSecurityList,
+  Credentials,
+  getOcpiTagString,
+  HttpMethod,
+  Namespace,
+  OcpiResponse,
+  OcpiTag,
+  SystemConfig,
+} from '@citrineos/base';
+import { OcpiModule } from './module';
+import { FastifyInstance, FastifyRequest } from 'fastify';
+import { ILogObj, Logger } from 'tslog';
+import { AuthorizationHeaderSchema } from './schema/authorization.header.schema';
+import { CredentialsService } from './service/credentials.service';
+import { CredentialsRepository } from './repository/credentials.repository';
+import { VersionRepository } from './repository/version.repository';
+import { VersionIdParam } from './schema/version.id.param.schema';
+import { targetConstructorToSchema } from 'class-validator-jsonschema';
+
+export class CredentialsModuleApi extends AbstractModuleApi<OcpiModule> {
+  private credentialsService: CredentialsService;
+
+  constructor(
+    config: SystemConfig,
+    module: OcpiModule,
+    server: FastifyInstance,
+    logger?: Logger<ILogObj>,
+    credentialsRepository?: CredentialsRepository,
+    versionRepository?: VersionRepository,
+  ) {
+    super(module, server, logger);
+
+    const finalCredentialsRepository =
+      credentialsRepository || new CredentialsRepository(config, this._logger);
+
+    const finalVersionRepository =
+      versionRepository || new VersionRepository(config, this._logger);
+
+    this.credentialsService = new CredentialsService(
+      this._logger,
+      finalCredentialsRepository,
+      finalVersionRepository,
+    );
+  }
+
+  @AsDataEndpoint(
+    '/ocpi/:versionId/credentials',
+    HttpMethod.Get,
+    undefined,
+    undefined,
+    targetConstructorToSchema(VersionIdParam),
```

Comment:
`targetConstructorToSchema` is used in @AsOcpiEndpoint as well but here, we need to do it manually as to not to infringe on existing functionality, although there may be a better way to make sure that the schema is generated closer to fastify. Additionally, with fastify not having a very robust setup with `class-validator` which is being used to generate the schema via the annotations - I have seen some of the annotations like `@Equals(VersionNumber.TWO_DOT_TWO_DOT_ONE)` fail in this setup because class validator isnt actually getting a chance to run because we are only using it to generate the schema via `targetConstructorToSchema` - @thanaParis I would love to use these annotations more extensively, as I think it would add value to our code base - what are your thoughts here, this does work as is - but can be improved...